CMakeLists.txt

```diff
@@ -1,41 +1,11 @@
-cmake_minimum_required (VERSION 3.1)
-project (Cas-OFFinder)
+cmake_minimum_required (VERSION 3.19.1)
```

Comment:
Is the newest version needed? I prefer keeping the oldest version here to make it compatible with old systems.

Reply:
I am sure the newest version is not needed, and I agree with the idea of preserving compatibility for old systems.  On the other hand, I do not know CMake well, so I am uncertain how far back it is possible to go (and have the improved modules work as required here).

TL;DR I will roll back to 3.14 and see what happens.  FWIW, I think it could use latest and it would not be too much of a problem.  More info below.

I have just read all the release notes between 3.1 and 3.19.

In 3.10, cmake itself dropped support for C++98 compilers.  (So if you are on a system which requires a C++98 compiler, then you cannot compile cmake >= 3.10.  Unlikely situation, I believe.)

In 3.14, cmake dropped support for WindowsXP and Vista.

In 3.15, cmake requires XCode 5 for the XCode generator.
  (This is not relevant here; I suppose anyone doing current macOS/iOS development is already > XCode5 based on Apple itself, but also, multiple cmake versions could be installed if < XCode 5 support is required for a different project.)

In 3.19, cmake dropped compatibility with build systems generates by cmake < 2.8.12.  (Not relevant here, and again an additional, old version of cmake could be installed by people with other, old cmake projects.  I do not think we need to worry about this.)

I think cmake support of old systems is quite good, and updating to latest cmake will not leave behind very many (maybe zero) users.

So, it probably comes down to "install convenience" as .msi / .exe, DMG, vendor .deb/.rpm.

Windows users will presumably download binaries from KitWare, so installing the latest version is easy for them.

macOS users can install the latest cmake easily via HomeBrew, and are probably accustomed to regular updates in any case.  (I myself find this frustrating though!)  HomeBrew cmake does not depend on any non-system packages, so updating cmake should not be a problem (i.e., no updates of other packages would be required).

macOS users can also download binaries from KitWare, support for macOS > 10.10.

Ubuntu Linux users can use KitWare's [APT repo](https://apt.kitware.com/), which supports Ubuntu 16.04, 18.04, 20.04.  This handles the case where Ubuntu itself does not update cmake.

For common platforms, this leaves Fedora Linux.  I am not sure how far back Fedora will issue updates, but I guess it is not very far since Fedora moves quickly.

Reply:
Then let's keep it 3.1 ATM. This is the oldest release supports `set (CMAKE_CXX_STANDARD 11)`.

Reply:
Ok, I will check if the changes I made work with 3.1.

I do think it would be good to update, at least partly.  There are lots of fixes in cmake on each release and it is a fairly complex build system.